src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const findUser = (userId: number) => (
+  usersFromServer.find(user => user.id === userId) || null
+);
+
+export const newTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(newTodos);
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [incorrectUser, setIncorrectUser] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+
+  const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const hastitleError = !title.trim();
+    const hasIncorrectUser = selectedUser === 0;
+
+    setTitleError(hastitleError);
+    setIncorrectUser(hasIncorrectUser);
+
+    if (hastitleError || hasIncorrectUser) {
+      return;
+    }
+
+    setTodos(prevTodos => {
+      const newId = Math.max(...prevTodos.map(todo => todo.id)) + 1;
```

Comment:
Create separate function for getting new id and move it to separate file `helpers`